src/sentry/web/frontend/oauth_token.py

```diff
@@ -46,81 +48,90 @@ def post(self, request: HttpRequest) -> HttpResponse:
         client_id = request.POST.get("client_id")
         if not client_id:
             return self.error(request, "invalid_client", "missing client_id")
-        if grant_type == GrantTypes.AUTHORIZATION:
-            return self._get_access_tokens(request, client_id)
-        elif grant_type == GrantTypes.REFRESH:
-            return self._get_refresh_token(request, client_id)
-        else:
+        if grant_type not in [GrantTypes.AUTHORIZATION, GrantTypes.REFRESH]:
             return self.error(request, "unsupported_grant_type")
-
-    def _get_access_tokens(self, request, client_id):
-        redirect_uri = request.POST.get("redirect_uri")
-        code = request.POST.get("code")
-
         try:
             application = ApiApplication.objects.get(
                 client_id=client_id, status=ApiApplicationStatus.active
```

Comment:
Actually - shouldn't we be utilizing the secret to fetch the apiapplication?
If someone requests with the right client_id, but wrong secret, we should just return 'apiapplication not found' - not necessarily "oh you have a right client_id, but wrong secret"

Reply:
@mdtro should we be deliberately opaque in our error messages here?

Reply:
I believe the accepted practice is to have identical errors for either the ID or secret being invalid, something like: `invalid client ID or secret`

Reply:
So it doesn't get lost in the Slack depths. 🙂 

> We should fetch with client_id + client_secret and return a general error that does not indicate which of the two is valid/invalid.
>
> If they send an invalid combination, I would suggest a response of 401 Unauthorized and an error message of invalid credentials.
>
>If they send just a client_id and no client_secret, respond with a 400 Bad Request with an error message about the client_secret  being required. Same for a missing client_id.

Reply:
fixed :)